File: DSA/Recursion/sum array/sumArray.js
#!/bin/node

/**
 *    This is an implementation of Recursive SumArray Function.
 *  Just for the Visualization and Study Purpose.
 */

const sumArray = (array, index) => {
  // The Base Case
  if (index === array.length - 1) return array[index];

  // Temp Variable holds the Sum
  let total = array[index] + sumArray(array, index + 1);

  return total;
};

const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
console.log(sumArray(nums, 0));

File: DSA/Recursion/sum array/sum_array.py
#!/bin/python3
"""
    This is an implementation of Recursive SumArray Function.
    Just for the Visualization and Study Purpose.
"""
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def sum_array(array, idx):
    """Solution."""
    # The Base Case.
    if idx == len(array) - 1:
        return array[idx]

    # The Recursive Functioning.
    return array[idx] + sum_array(array, idx + 1)

print(sum_array(nums, 0))
